src/types.rs

```diff
@@ -1154,7 +1154,7 @@ macro_rules! from_redis_value_for_tuple {
                     return Ok(rv)
                 }
                 let mut offset = 0;
-                while offset < items.len() - 1 {
+                while offset < items.len() {
```

Comment:
```rust
                for chunk in items.chunks_exact(n) {
                    let mut offset = 0;
                    rv.push(($({let $name = (); from_redis_value(
                         &chunk[{ offset += 1; offset - 1 }])?},)*));
                }
```

Reply:
Update the example a bit more @Terkwood

Reply:
I had to massage the syntax on this a little bit, and it generates some pretty crazy errors.  I caught one of your previous iterations in the comments and those seemed a bit more benign 

```text
error[E0308]: mismatched types
    --> src/types.rs:1159:50
     |
1159 |                          [$($name),*] => rv.push(($(from_redis_value($name)),*),
),
     |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
expected tuple, found enum `std::result::Result`
...
1182 | from_redis_value_for_tuple! { T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12,
 }
     | --------------------------------------------------------------------------------
-- in this macro invocation
     |
     = note: expected tuple `(T12,)`
                 found enum `std::result::Result<_, types::RedisError>`
     = note: this error originates in a macro (in Nightly builds, run with -Z macro-bac
ktrace for more info)

error: aborting due to 78 previous errors
```

Reply:
Currently using

```rust
for chunk in items.chunks_exact(n) {
                    match chunk {
                         [$($name),*] => rv.push(($(from_redis_value($name)),*),),
                         _ => unreachable!(),
                     }
                }
```

Reply:
`if let Ok(a)` to grab those inner values from the `$(from_redis_value($name))` might work fine, I'm just trying to tread lightly here :grin:

Reply:
I think that should be (missing `?` to unpack the results)

```
for chunk in items.chunks_exact(n) {
    match chunk {
        [$($name),*] => rv.push(($(from_redis_value($name)?),*),),
         _ => unreachable!(),
    }
}
```

Reply:
Just got this version pushed, and the tests are running now in Travis &c.  :sunglasses: